src/argilla/client/feedback/training/schemas.py

```diff
@@ -356,6 +413,47 @@ def __label2id__(self):
     def __id2label__(self):
         return self.label.question.__id2label__
 
+    def _format_data(self, dataset: "FeedbackDataset"):
+        if self.formatting_func is not None:
+            output = set()
+            for sample in dataset.format_as("datasets"):
+                text_label = self.formatting_func(sample)
+                if text_label is None:
+                    continue
+
+                if isinstance(text_label, tuple) and isinstance(text_label[0], str):
+                    text_label = {text_label}
+                else:
+                    raise ValueError(
+                        "formatting_func must return (text,label) as a Tuple[str, str] or a Tuple[str, List[str]]"
+                    )
+
+                output |= set(text_label)
+
+            data = []
+            _all_labels = set()
+            for text, label in output:
+                data.append({"text": text, "label": label})
+                if isinstance(label, list):
```

Comment:
```suggestion
                if isinstance(label, list):
```
Would fail for tuples, is that a problem?